sdk-setup/src/mb2

```diff
@@ -1791,20 +1791,88 @@ run_build_requires__process_args() {
         BUILD_REQUIRES_MODE=pull
     fi
 }
+
 sign_rpms() {
     [[ -e $STATEDIR/spec ]] || fatal "No previous build found"
     local rpms=$(find_rpms $all_opt "$OPT_OUTPUTDIR" "$STATEDIR/spec")
     [[ $rpms ]] || fatal "No package found"
+
     local sdksign=/usr/libexec/sdk-setup/sdk-rpmsig
+
     if [[ -x $sdksign ]]; then
         $sdksign ${rpms}
     else
-        rpmsign --addsign ${rpms}
+        sign_rpms_native ${rpms}
     fi
+
     local rc=$?
     [[ $rc -eq 0 ]] || { notice "Failed to sign packages"; return $rc; }
 }
 
+sign_rpms_native() {
+    local rpms=("$@")
+
+    local tmpdir=
+    sign_rpms_native_cleanup() (
+        trap 'echo cleaning up...' INT TERM HUP
+        if [[ $tmpdir ]]; then
+            rm -rf "$tmpdir"
+        fi
+    )
+    trap 'sign_rpms_native_cleanup; trap - RETURN' RETURN
+    trap 'return 1' INT TERM HUP
+
+    local extra_args=()
+
+    if [[ $OPT_PACKAGE_SIGN_USER ]]; then
+        if ! gpg2 --list-secret-keys "$OPT_PACKAGE_SIGN_USER" &>/dev/null; then
+            #if there is no known key in gpg2, try to import it from a file
+            key_path="${OPT_SHARED_DIR}/gnupg/${OPT_PACKAGE_SIGN_USER}.key"
+            if [[ -f $key_path ]]; then
+                if ! gpg2 --import "${key_path}" &> /dev/null; then 
+                    notice "Cannot sign packages: Failed to import key from file '$key_path'."
+                    return 1
+                fi
+            else
+                notice "Cannot sign packages: No GPG key found for the user '$OPT_PACKAGE_SIGN_USER' and no GPG key file '$key_path'."
+                return 1
+            fi
+        fi
```

Comment:
We should be specific about the import functionality being limited to build engine and we also should try harder - what if the user creates a new key pair using the same name? We also cannot simply import the new key without removing the old one first - we would end up with multiple keys matching the user name under the build engine and how would rpmsign tell which one to use?

```suggestion
        if inside_build_engine; then
            local shared_key=${OPT_SHARED_DIR}/gnupg/${OPT_PACKAGE_SIGN_USER}.key
            if [[ ! -f $shared_key ]]; then
                notice "Cannot sign packages: Internal error: No GPG key shared for the user '$OPT_PACKAGE_SIGN_USER'."
                return 1
            fi

            ... import new / update existing ...
        fi
        
        if ! gpg2 --list-secret-keys "$OPT_PACKAGE_SIGN_USER" &>/dev/null; then
            notice "Cannot sign packages: No GPG key found for the user '$OPT_PACKAGE_SIGN_USER'."
            return 1
        fi
```

Reply:
And this is the reason why I didn't want to support import in the signature function. It's not obvious. Why, if there are no key files, we should log out and not sign what is already in the system with this name?

Reply:
Application SDK users use QtC (sfdk) and the build engine is a black box to them. They have told QtC (sfdk) which key should be used for signing packages and that's all they need to care about. It is QtC's (sfdk's) responsibility to pass the keys to mb2 and if it fails to, it is an internal error and mb2 has no option to recovery from that.